packages/devtools_app/integration_test/test/live_connection/debugger_panel_test.dart

```diff
@@ -0,0 +1,155 @@
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  late TestApp testApp;
+
+  setUpAll(() {
+    testApp = TestApp.fromEnvironment();
+    expect(testApp.vmServiceUri, isNotNull);
+  });
+
+  testWidgets('Debugger panel', (tester) async {
+    await pumpAndConnectDevTools(tester, testApp);
+    await switchToScreen(tester, ScreenMetaData.debugger);
+    await tester.pump(safePumpDuration);
+
+    logStatus(
+      'looking for the main.dart file',
+    );
+
+    // Look for the main.dart file name:
+    expect(find.text('package:flutter_app/main.dart'), findsOneWidget);
+
+    // Look for the main.dart source code:
+    final line1LineFinder = find.byKey(const Key('Line Item 1'));
+    expect(line1LineFinder, findsOneWidget);
+    expect(getSourceCodeAtLine(line1LineFinder), contains('FILE: main.dart'));
+
+    logStatus(
+      'opening the "more" menu',
+    );
+
+    final moreMenuFinder = find.byType(PopupMenuButton<ScriptPopupMenuOption>);
+    expect(moreMenuFinder, findsOneWidget);
+    await tester.tap(moreMenuFinder);
+    await tester.pumpAndSettle(safePumpDuration);
+
+    logStatus(
+      'selecting the go-to-line menu option',
+    );
+
+    final goToLineOptionFinder = find.textContaining('Go to line number');
+    expect(goToLineOptionFinder, findsOneWidget);
+    await tester.tap(goToLineOptionFinder);
+    await tester.pumpAndSettle(safePumpDuration);
+
+    logStatus(
+      'entering line number in the go-to-line dialog',
+    );
+
+    final goToLineInputFinder = find.widgetWithText(TextField, 'Line Number');
+    expect(goToLineInputFinder, findsOneWidget);
+    await tester.enterText(goToLineInputFinder, '24');
+    await tester.testTextInput.receiveAction(TextInputAction.done);
+    await tester.pumpAndSettle(safePumpDuration);
+
+    logStatus(
+      'looking for line 24',
+    );
+
+    // Look for the line 24 gutter item:
+    final line24GutterFinder = find.byKey(const Key('Gutter Item 24'));
+    expect(line24GutterFinder, findsOneWidget);
+
+    // Look for the line 24 line item:
+    final line24LineFinder = find.byKey(const Key('Line Item 24'));
+    expect(line24LineFinder, findsOneWidget);
+    expect(getSourceCodeAtLine(line24LineFinder), contains('count++;'));
+
+    logStatus(
+      'setting a breakpoint',
+    );
+
+    // Tap on the gutter for the line to set a breakpoint:
+    await tester.tap(line24GutterFinder);
+    await tester.pumpAndSettle(safePumpDuration);
+
+    logStatus(
+      'pausing at breakpoint',
+    );
+
+    final frameFinder = findStackFrameWithText('PeriodicAction.doEvery');
+    expect(frameFinder, findsOneWidget);
+    expect(isLineFocused(line24LineFinder), isTrue);
+
+    logStatus(
+      'inspecting variables',
+    );
+
+    final countVariableFinder = find.textContaining('count:');
+    expect(countVariableFinder, findsOneWidget);
+
+    logStatus(
+      'switching stackframes',
+    );
+
+    // Tap on the stackframe:
+    await tester.tap(frameFinder);
```

Comment:
Are we supposed to be switching to a different stack frame than the one we are already on?

Reply:
We are! Goal is to catch https://github.com/flutter/devtools/issues/5707